### PR TITLE
Added modules for repo sync

### PR DESCRIPTION
**Pull request by Akhilesh2412**
Added get status module and sync repo module

### CODE REVIEW

File: library/nsxt_repo_sync.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+DOCUMENTATION = '''
+---
+module: nsxt_repo_sync
+short_description: 'Synchronizes the repository data between nsx managers'
+description: "Attempts to synchronize the repository partition on nsx 
+              manager. Repository partition contains packages required for 
+              the install and upgrade of nsx components.Normally 
+              there is no need to call this API explicitely by the user."
+version_added: '2.7'
+author: 'Rahul Raghuvanshi'
```
Comment (r-raghu) [inline library/nsxt_repo_sync.py]: Change this to your name, as this is complete new module written by you.

File: library/nsxt_repo_sync.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+# BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import absolute_import, division, print_function
```
Comment (r-raghu) [inline library/nsxt_repo_sync.py]: Where are you using division and print_funtion?

File: library/nsxt_repo_sync.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 VMware, Inc.
```
Comment (r-raghu) [inline library/nsxt_repo_sync.py]: Change copyright to 2019

File: library/nsxt_repo_sync_facts.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 VMware, Inc.
```
Comment (r-raghu) [inline library/nsxt_repo_sync_facts.py]: Change copyright to 2019

File: library/nsxt_repo_sync_facts.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+# BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import absolute_import, division, print_function
```
Comment (r-raghu) [inline library/nsxt_repo_sync_facts.py]: Where are you using division and print_funtion?

File: library/nsxt_repo_sync_facts.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: nsxt_repo_sync_facts
+short_description: 'Synchronizes the repository data between nsx managers.'
```
Comment (r-raghu) [inline library/nsxt_repo_sync_facts.py]: Correct short description

File: module_utils/common_utils.py [hunk 7]
```diff
@@ -26,6 +26,19 @@ def check_if_valid_ip(ip_address):
...
+    '''
+    params:
+    - object: Object where value is to be searched from attribute list
+    - attribute_list: List to be used for searching attribute value
+    '''
+    for attribute in attribute_list:
+        if object.__contains__(attribute):
+            object = object[attribute]
+        else:
+            module.fail_json(msg='Error while  traversing and retrieving value. Attribute named %s was not found.'% attribute)
```
Comment (r-raghu) [inline module_utils/common_utils.py]: raise exception and catch in parent function.

Comment (Akhilesh2412) [reply module_utils/common_utils.py]: Done.

File: library/nsxt_repo_sync_facts.py (hunk 6 above)
Comment (Akhilesh2412) [reply library/nsxt_repo_sync_facts.py]: Done.

File: library/nsxt_repo_sync_facts.py (hunk 5 above)
Comment (Akhilesh2412) [reply library/nsxt_repo_sync_facts.py]: Not done.

File: library/nsxt_repo_sync_facts.py (hunk 4 above)
Comment (Akhilesh2412) [reply library/nsxt_repo_sync_facts.py]: Done

File: library/nsxt_repo_sync.py (hunk 3 above)
Comment (Akhilesh2412) [reply library/nsxt_repo_sync.py]: Done

File: library/nsxt_repo_sync.py (hunk 1 above)
Comment (Akhilesh2412) [reply library/nsxt_repo_sync.py]: Done.

**Pull request closed without merging**